### PR TITLE
Added termnial-menu and refactored a lot

### PR DESCRIPTION
**Pull request by antonericson**

### CODE REVIEW

File: automatic_generation.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import tts
+
+BACKGROUND_VIDEO_PATH = './video-generation/public/video/backgroundVideo.mp4'
+OUTPUT_FILE_BUZZ_WORDS = 'reddit_story_crazy_reading_reddit_short_funny_sad_insane'
+RUN_OPTIONS = ['One', 'Specify number']
+
+def preview():
+    generate_videos(number_of_videos=1, run_in_preview_mode=True)
+
+def default():
```
Comment (Jaagarn) [inline automatic_generation.py]: I assume default only runs after you have selected 1 or 0, and you can't select anything else?

File: automatic_generation.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    terminal_menu = TerminalMenu(RUN_OPTIONS)
+    selected_index = terminal_menu.show()
+
+    if selected_index == 0:
+        number_of_videos = 1
+    elif selected_index == 1:
+        number_of_videos = int(input('Input number of videos to generate:\n'))
```
Comment (Jaagarn) [inline automatic_generation.py]: And if I give the input "I'm very difficult and wont put an integer here"?

File: automatic_generation.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    tts_instance = TTS(model_name="tts_models/en/vctk/vits")
+
+    while generated_videos < number_of_videos :
+        # Get a reddit post
+        try:
+            post = ri.get_post()
+        except ImportError as error:
+            log.warning(error)
+            return
+        title = post['title']
```
Comment (Jaagarn) [inline automatic_generation.py]: I don't have anything against it, but why are we not just sending the full post object to the different methods?

File: automatic_generation.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        generate_video_command = f'cd video-generation; npx remotion render RedditStory\
+            ../out/{subreddit}_{post_id}/{subreddit}_{OUTPUT_FILE_BUZZ_WORDS}.mp4 --props=../current-config.json'
+        preview_video_command = 'cd video-generation; npx remotion preview RedditStory --props=../current-config.json'
+
+        if not run_in_preview_mode:
+            subprocess.run(generate_video_command, shell=True, check=False)
+        else:
+            subprocess.run(preview_video_command, shell=True, check=False)
+
+        # Clean up temporary files
```
Comment (Jaagarn) [inline automatic_generation.py]: ❤️

File: interactive_mode.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        post = ri.get_post(allow_over_18=True)
+    except ImportError as error:
+        log.warning(error)
+        get_posts(videos_to_generate)
+
+    subreddit = post['subreddit']
+    title = post['title']
+    body = post['selftext']
+    is_nsfw = post['over_18']
+    nsfw_string = ''
```
Comment (Jaagarn) [inline interactive_mode.py]: nsfw_string= 'NSFW' if is_nsfw else ''

File: interactive_mode.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+                                     --terminal-width 200 current_post_tmp.md',
+                                 preview_size=0.75,
+                                 clear_screen=True)
+    selected_index = terminal_menu.show()
+
+    utils.remove_file('current_post_tmp.md')
+
+    if selected_index == 0 or selected_index == 2:
+        videos_to_generate.append(post)
+
```
Comment (Jaagarn) [inline interactive_mode.py:44]: extra enter

**Review by Jaagarn — APPROVED** [review]
"Approved with suggestions" is what I wanted to select. I think you could improve on some things, but it's not a requirement for approval

File: interactive_mode.py (hunk 5 above)
Comment (antonericson) [reply interactive_mode.py]: Yes, either it says NSFW or nothing in the preview.

Comment (Jaagarn) [reply interactive_mode.py]: int b = isBFive ? 5 : 0;

**Pull request merged**